### ISSUE TITLE
Add aggregated map page?

### ISSUE THREAD

**Issue by bondjimbond** [Islandora-Labs/islandora_simple_map#10]
Would be neat to generate a page that generates a map of all Islandora objects with location information, with a Solr query.

**Comment by mjordan** [comment]
@bondjimbond yes that would be neat. The simple API that this module uses wouldn't support that, but it sounds like something that could be accomplished using the full Maps API as described in #7. 

https://github.com/jyobb/islandora_map_browse already provides this sort of thing, but not using Google Maps.

**Comment by whikloj** [comment]
@bondjimbond this sort of fits in with my [collection use case](https://github.com/mjordan/islandora_simple_map/issues/17), perhaps we need something along the lines of the [hook](https://github.com/mjordan/islandora_simple_map/issues/18) I am implementing to allow different coordinate systems.

**Comment by mjordan** [comment]
@bondjimbond can you provide more detail? Thanks to @whikloj we now have collection maps that are populated using a Solr query. Are you wanting to be able to use an arbitrary Solr query to create a map (e.g. for a particular place name, etc.)?

**Comment by whikloj** [comment]
@mjordan my reading of @bondjimbond's request here was a request of **ALL** of the items in your repository on a single map.

I have heard discussion of the same request from my institution, however it would require your items all having some sort of location to ensure they appear on the map or perhaps some sort of mods:hierarchicalGeographic -> coordinate parser.

It would also probably require the use of the [marker clustering](https://developers.google.com/maps/documentation/javascript/marker-clustering) library to help with keeping the map readable.

But I guess I am not sure what you get at the end, does it give you search results or individual links to resources?

**Comment by MarcusBarnes** [comment]
I'd suggest that this use case is outside the scope of Simple Map.   There are other mapping modules for Drupal 7 that can be used with an appropriately crafted Solr view to produce what is suggested in @whikloj's [comment above](https://github.com/mjordan/islandora_simple_map/issues/10#issuecomment-379748440).

**Comment by whikloj** [comment]
@MarcusBarnes can you elaborate a bit on this, especially about which modules should be used in favour of this one.

**Comment by MarcusBarnes** [comment]
Here's [an example](http://culinaria.digitalscholarship.utsc.utoronto.ca/mapping-scarborough-chinatown) of what I think you're suggesting from a purely Drupal 7 site.  Zoom out and you'll see the clustering.  I'll have to ask what the specific Drupal 7 mapping modules are.   From my understanding, there's a Solr view that pulls the geo-coordinates, then the mapping module is able to present the results.

My suggestion is that using an appropriately crafted Islandora Solr View query and similar Drupal 7 mapping modules, you should be able to get a similar result (with some work).

**Comment by whikloj** [comment]
This is my guess at the underlying module of the above site.
https://www.drupal.org/project/ip_geoloc

**Comment by mjordan** [comment]
I'm having trouble with large collections. The one I am testing with has 5142 members that have geocoordinates. The map never renders; what is strange is that the server also doesn't time out. Therefore I suspect it's the Javascript that populates the map that hits a wall. I will continue to debug.

I am leaving this comment here since generating a map that has lots of points on it might be problematic. What's the largest map (in terms of points) anyone has encountered?

**Comment by whikloj** [comment]
@mjordan neat I have never tried that many objects, but we should open a new issue as it will be related to a mistake in my cod and not to this request.

**Comment by mjordan** [comment]
@whikloj will do, and try to provide more data.

**Comment by mjordan** [comment]
It's the looping through Solr results with a page size of [20](https://github.com/mjordan/islandora_simple_map/blob/7.x/includes/utilities.inc#L352). I changed it to 1000000 (which is the 'rows' value I use when I want all possible records in one query) and the map renders, but only with < 20 points due to the subsequent `while` loop, I believe. I'll open a ticket later today.
